### DISCUSSION TITLE
Github webhooks not working - Response "405 - Method Not Allowed"

### DISCUSSION THREAD

**Discussion by vvscloud** [buildbot/buildbot#7556] [Q&A] [answered]
We have configured our buildbot and enabled github webhooks on http://\<server-name\>:8010/#/change_hook/github
Selected pull requests.

In our master.cfg:
`c['www'] = {
    "port": 8010,
    "change_hook_dialects": {
        #'base': True,
        'github': {
            'github_api_endpoint': ''<>",
            'token': '<>',
            'secret':'<>'
        },
    },
    "plugins": dict(waterfall_view={}, console_view={}, grid_view={})
}`

Once we enable webhooks - we get 

`
    <html>
      <head><title>405 - Method Not Allowed</title></head>
      <body>
        <h1>Method Not Allowed</h1>
        <p>Your browser approached me (at /) with the method "POST".  I only allow the methods HEAD, GET here.</p>
      </body>
    </html>
    
    Allow: HEAD, GET
    Content-Length: 233
    Content-Type: text/html; charset=utf-8
`
Please help on this, Why POST methods are not enabled?

**Comment by vvscloud** [comment]
Any help here?

**Reply by p12tic** [reply, marked as answer]
Could it be that hook URL should be `http://<server-name>:8010/change_hook/github` (note the omission of `#`).

**Reply by vvscloud** [reply]
I have tried it. Then it returns none. 
With # it returns atleast 405.

I can see in http.log , the recieving of post hit from GitHub but it did not respond back. 

Not sure what I am missing here :(

**Reply by vvscloud** [reply]
Is there a way to check if POST is enabled in buildbot server? 
I do not see any logs of dialects as well.

**Reply by tdesveaux** [reply]
You should see some logs in `twistd.log`.

Could you share your change_hook configuration on both Buildbot and Github's side?

**Reply by vvscloud** [reply]
@p12tic  - you are right :) keeping # caused the change in behavior.
Thank you everyone for helping me on this!
